### ISSUE TITLE
modal has incorrect aria role

### ISSUE THREAD

**Issue by gebhardtr** [carbon-design-system/carbon#4557]
## Environment
> Automated testing tool and ruleset

[pa11y-ci](https://github.com/pa11y/pa11y-ci) v2.3.0 w/ default ruleset

## Detailed description

> What version of the Carbon Design System are you using?

carbon-components@10.6.1
carbon-components-react@7.6.1 

> What did you expect to happen?

No errors from pa11y, aria role=dialog on ModalWrapper and no role on children.

> What happened instead?

Pa11y tests failed.

> What WCAG 2.1 checkpoint does the issue violate?

WCAG2AA.Principle1.Guideline1_3.1_3_1.F92,ARIA4 (https://www.w3.org/TR/wai-aria-1.0/roles#presentation)

## Steps to reproduce the issue

1. Install pa11y-ci
2. Create and wrap a component that embeds Modal and toggles it through state
```javascript
<Modal
  className='some-class'
  aria-label='label'
  open={this.state.modalIsOpen}
  onRequestClose={this.modalClose}
  passiveModal
  modalLabel='label'
  modalHeading='heading>
    <Child />
</Modal>
```
3. Start app that renders component
4. Run pa11y-ci against app/path-that-renders-modal

**Comment by dakahn** [comment]
Hey, thanks for submitting. We're not seeing this as a problem in our a11y testing here. Not sure why Pa11y would pick up this role on our presentational _background_ element -- the modal itself if you inspect the code actually has `role="dialog"`

Did you have more insight into how this is an accessibility problem?

**Comment by abbeyhrt** [comment]
We aren't sure why Pa11y is picking this up as an error but we've looked into it and will keep the `role="presentation"` for our modal wrapper since that it how we are using it. Closing this but feel free to get in touch if you think something hasn't been addressed!
